### ISSUE TITLE
BUG TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type object

### ISSUE THREAD

**Issue by davidmwhynot** [netlify/zip-it-and-ship-it#32]
```
> netlify deploy
Deploy path:        D:\xampp\htdocs\netlify_sandbox\dist
Functions path:     D:\xampp\htdocs\netlify_sandbox\functions
Configuration path: D:\xampp\htdocs\netlify_sandbox\netlify.toml
Deploying to draft URL...
- Hashing files...
 »   Warning:
 »   {}
 »
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type object
    at filesForFunctionZip (C:/Users/david/AppData/Roaming/npm/node_modules/netlify-cli/node_modules/@netlify/zip-it-and-ship-it/src/zip.js:38:22)
    at zipJs (C:/Users/david/AppData/Roaming/npm/node_modules/netlify-cli/node_modules/@netlify/zip-it-and-ship-it/src/zip.js:110:3)
    at zipFunction (C:/Users/david/AppData/Roaming/npm/node_modules/netlify-cli/node_modules/@netlify/zip-it-and-ship-it/src/zip.js:143:12)
    at C:/Users/david/AppData/Roaming/npm/node_modules/netlify-cli/node_modules/@netlify/zip-it-and-ship-it/src/zip.js:181:9
    at element (C:/Users/david/AppData/Roaming/npm/node_modules/netlify-cli/node_modules/p-all/index.js:4:67)
    at Promise.resolve.then.element (C:/Users/david/AppData/Roaming/npm/node_modules/netlify-cli/node_modules/p-map/index.js:47:21)
```
## specs
windows 10
```
> node -v
v11.1.0
> npm -v
6.4.1
```

**Comment by davidmwhynot** [comment]
Fixed by running `npm init`. Apparently not having a package.json file was causing something to break.

**Comment by sw-yx** [comment]
indeed we have had this bug reported internally as well. we will put a nicer error.

**Comment by bcomnes** [comment]
I think I see the bug in ZISI.  Writing some tests now.

**Comment by bcomnes** [comment]
Fixed in `netlify-cli@2.11.12`.

**Comment by oles** [comment]
@bcomnes I have no words.
I had the same error.

I tried so many different things.
Was about to give up, after I saw this, and as a last resort, copied a random package.json to my project root.

It worked.

I'm running version 2.37.0 of the cli.

I need a break.
